pkg/authn/keychain.go

```diff
@@ -35,7 +35,19 @@ type Keychain interface {
 
 // defaultKeychain implements Keychain with the semantics of the standard Docker
 // credential keychain.
-type defaultKeychain struct{}
+type defaultKeychain struct {
+	configDir string
```

Comment:
Add comment saying this will store a custom docker config directory. The convention in Go is
`// configDir will store a custom docker client config directory. If this is blank, config directory resolution will mimic docker's default behavior`